core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/configuration/security/jwt/JWTAuthenticationFilterTest.java

```diff
@@ -0,0 +1,82 @@
+package io.metadew.iesi.server.rest.configuration.security.jwt;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.mock.web.MockFilterChain;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@WebMvcTest(JWTAuthenticationFilter.class)
+@ContextConfiguration(classes = {JWTAuthenticationFilter.class, JWTAuthenticationConverter.class, JwtService.class})
+@ActiveProfiles({"test", "security"})
+
+public class JWTAuthenticationFilterTest {
+
+    @Autowired
+    private JWTAuthenticationConverter jwtAuthenticationConverter;
+
+    @Test
+    public void testJWTDecodeException() throws Exception {
+
+        String token = "eyJ0eXAiOJKV1QiLCUzI1NiJ9.eyJzdWIiOiJMyIiwiaXNzIjoiaWVzaSIsImV4cCI6MTU5ODU2ODM2MywiaWF0IjoxNTk4NDY4MzYzLCJhdXRob3JpdGllcyI6W119.alYmNwRyiqS8VijXdQ2F0ooVdr5KkITz2bycffG6_fI";
+        MockHttpServletRequest request = new MockHttpServletRequest("GET", "/scripts");
+        request.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+        JWTAuthenticationFilter jwtAuthenticationFilter = new JWTAuthenticationFilter(jwtAuthenticationConverter);
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        MockFilterChain chain = new MockFilterChain();
+
+        jwtAuthenticationFilter.doFilterInternal(request, response, chain);
+        assertThat(response.getStatus()).isEqualTo(401);
+        assertThat(response.getContentAsString()).contains("com.auth0.jwt.exceptions.JWTDecodeException: The string");
```

Comment:
is it possible to do a jsonPath assertion on the json string? As can you check that the 'message' field starts with "com.auth0..." and the 'status' field is equal to 401.